src/dotnet/src/Mil.Navy.Nrl.Norm/NormObject.cs

```diff
@@ -0,0 +1,177 @@
+﻿using System.Runtime.InteropServices;
+
+namespace Mil.Navy.Nrl.Norm
+{
+    /// <summary>
+    /// The base transport object.
+    /// </summary>
+    public class NormObject
+    {
+        /// <summary>
+        /// The special value NORM_OBJECT_INVALID corresponds to an invalid reference.
+        /// </summary>
+        public const int NORM_OBJECT_INVALID = 0;
+        /// <summary>
+        /// Used to reference state kept for data transport objects being actively transmitted or received.
+        /// </summary>
+        protected long _handle;
+
+        /// <summary>
+        /// Internal constructor for NormObject
+        /// </summary>
+        /// <param name="handle">Identifies an instance of NormObject</param>
+        internal NormObject(long handle)
+        {
+            _handle = handle;
+        }
+
+        /// <summary>
+        /// Used to reference state kept for data transport objects being actively transmitted or received.
+        /// </summary>
+        public long Handle => _handle;
+
+        /// <summary>
+        /// Data associated with NormObject.
+        /// </summary>
+        public byte[]? Info
+        {
+            get
+            {
+                if (!NormObjectHasInfo(_handle))
+                {
+                    return null;
+                } 
+
+                var length = NormObjectGetInfoLength(_handle);
+                var buffer = new byte[length];
+                var bufferHandle = GCHandle.Alloc(buffer, GCHandleType.Pinned);
```

Comment:
The handle is not needed. You can use an `[Out] byte[]` in the PInvoke signature.